packages/administration/assets/src/js/utils/passwordStrengthCalculator.js

```diff
@@ -0,0 +1,58 @@
+export default class PasswordStrengthCalculator {
+    static STRENGTH_VERY_WEAK = 'VERY_WEAK';
+    static STRENGTH_WEAK = 'WEAK';
+    static STRENGTH_MEDIUM = 'MEDIUM';
+    static STRENGTH_STRONG = 'STRONG';
+    static STRENGTH_VERY_STRONG = 'VERY_STRONG';
+
+    // Entropy calculation based on the Symfony Password Strength Estimator
+    static calculate(password) {
+        const length = password.length;
+
+        if (!length) {
+            return this.STRENGTH_VERY_WEAK;
+        }
+
+        const charCounts = {};
+        for (const char of password) {
+            const code = char.charCodeAt(0);
+            charCounts[code] = (charCounts[code] || 0) + 1;
+        }
+
+        const chars = Object.keys(charCounts).length;
+
+        let control = 0,
+            digit = 0,
+            upper = 0,
+            lower = 0,
+            symbol = 0,
+            other = 0;
+
+        for (const chrStr in charCounts) {
+            const chr = parseInt(chrStr, 10);
+
+            if (chr < 32 || chr === 127) {
+                control = 33;
+            } else if (48 <= chr && chr <= 57) {
+                digit = 10;
+            } else if (65 <= chr && chr <= 90) {
+                upper = 26;
+            } else if (97 <= chr && chr <= 122) {
+                lower = 26;
+            } else if (128 <= chr) {
+                other = 128;
+            } else {
+                symbol = 33;
+            }
+        }
+
+        const pool = lower + upper + digit + symbol + control + other;
+        const entropy = chars * Math.log2(pool) + (length - chars) * Math.log2(chars);
+
+        if (entropy >= 120) return this.STRENGTH_VERY_STRONG;
+        if (entropy >= 100) return this.STRENGTH_STRONG;
+        if (entropy >= 80) return this.STRENGTH_MEDIUM;
+        if (entropy >= 60) return this.STRENGTH_WEAK;
+        return this.STRENGTH_VERY_WEAK;
```

Comment:
This is quite unusual formatting, however, if it passes our standards, then :shrug:

Reply:
We could write it like 

```
switch (true) {
    case entropy >= 120: return this.STRENGTH_VERY_STRONG;
    case entropy >= 100: return this.STRENGTH_STRONG;
    case entropy >= 80:  return this.STRENGTH_MEDIUM;
    case entropy >= 60:  return this.STRENGTH_WEAK;
    default:             return this.STRENGTH_VERY_WEAK;
}
```

Is it better?

Reply:
Yeah, I like the switch better when I see it (I have learned not to dislike the `switch(true)` approach some time ago :smile: ). However, I did not have the switch in mind when writing the comment, I was referring just to the inline returns and the missing empty line before the last one. I would just expect something like this:

```
if (entropy >= 120) {
    return this.STRENGTH_VERY_STRONG;
}

if (...) {
// ...

return this.STRENGTH_VERY_WEAK;
```

Reply:
I used the switch instead. Full ifs seem too long for me